android/capacitor/src/main/java/com/getcapacitor/Plugin.java

```diff
@@ -283,47 +375,116 @@ public boolean hasRequiredPermissions() {
+        }
+
+        // default to base permission launcher that returns permission states via the plugin call
+        if (activityResultLauncher == null) {
+            activityResultLauncher = basePermissionLauncher;
+        }
+
+        CapacitorPlugin annotation = handle.getPluginAnnotation();
+        if (annotation != null) {
+            HashSet<String> perms = new HashSet<>();
+            for (Permission perm : annotation.permissions()) {
+                perms.addAll(Arrays.asList(perm.strings()));
+            }
+
+            bridge.savePermissionCall(call);
+            activityResultLauncher.launch(perms.toArray(new String[0]));
+        }
     }
 
     /**
-     * Helper for requesting specific permissions
-     *
-     * @since 3.0.0
-     * @param call the plugin call
-     * @param permissions the set of permissions to request
-     * @param requestCode the requestCode to use to associate the result with the plugin
+     * Request permissions using an alias defined on the plugin
+     * @param alias an alias defined on the plugin
+     * @param call the plugin call involved in originating the request
      */
-    public void requestPermissions(PluginCall call, String[] permissions, int requestCode) {
-        bridge.savePermissionCall(call);
-        ActivityCompat.requestPermissions(getActivity(), permissions, requestCode);
+    protected void requestPermissionForAlias(@NonNull String alias, @NonNull PluginCall call) {
+        requestPermissionForAliases(new String[] { alias }, call);
     }
 
     /**
-     * Request all of the specified permissions in the CapacitorPlugin annotation (if any)
-     *
-     * @since 3.0.0
-     * @param call the plugin call
+     * Request permissions using aliases defined on the plugin
+     * @param aliases a set of aliases defined on the plugin
+     * @param call the plugin call involved in originating the request
+     */
+    protected void requestPermissionForAliases(@NonNull String[] aliases, @NonNull PluginCall call) {
+        String callingMethodName = call.getMethodName();
+        ActivityResultLauncher<String[]> activityResultLauncher = permissionLaunchers.get(callingMethodName);
+        if (activityResultLauncher == null && !callingMethodName.equals("requestPermissions")) {
+            Logger.error(
+                String.format(
+                    Locale.US,
+                    "There is no permission callback method registered for the plugin method %s. " +
+                    "Please define a permissionCallback method name in the annotation and provide a " +
+                    "method that has the correct signature: (PluginCall, Map<String, PermissionState>)",
+                    callingMethodName
+                )
+            );
+        }
+
+        // default to base permission launcher that returns permission states via the plugin call
+        if (activityResultLauncher == null) {
+            activityResultLauncher = basePermissionLauncher;
+        }
```

Comment:
Can this logic be put into a private method for reuse? (it looks the same as in `requestAllPermissions`)